liblangutil/ErrorReporter.cpp

```diff
@@ -97,27 +100,29 @@ bool ErrorReporter::checkForExcessiveErrors(Error::Type _type)
 	{
 		m_warningCount++;
 
-		if (m_warningCount == c_maxWarningsAllowed)
+		if (m_warningCount - m_ignoreWarningCount == c_maxWarningsAllowed + 1)
 		{
 			auto err = make_shared<Error>(Error::Type::Warning);
 			*err << util::errinfo_comment("There are more than 256 warnings. Ignoring the rest.");
 			m_errorList.push_back(err);
 		}
 
-		if (m_warningCount >= c_maxWarningsAllowed)
+		if (m_warningCount - m_ignoreWarningCount >= c_maxWarningsAllowed + 1)
 			return true;
 	}
 	else
 	{
 		m_errorCount++;
 
-		if (m_errorCount > c_maxErrorsAllowed)
+		if (m_errorCount == c_maxErrorsAllowed + 1)
 		{
 			auto err = make_shared<Error>(Error::Type::Warning);
-			*err << util::errinfo_comment("There are more than 256 errors. Aborting.");
+			*err << util::errinfo_comment("There are more than 256 errors. Ignoring the rest.");
```

Comment:
I think we should really abort instead of just ignoring.

Reply:
And why?

Reply:
The idea behind this was that continuing the compilation just eats resources needlessly.